evap/results/views.py

```diff
@@ -179,15 +238,31 @@ def evaluation_detail(request, semester_id, evaluation_id):
         general_questionnaire_results_top += general_questionnaire_results_bottom
         general_questionnaire_results_bottom = []
 
-    evaluation.distribution = calculate_average_distribution(evaluation)
-    evaluation.avg_grade = distribution_to_grade(evaluation.distribution)
+    all_evaluations_of_course = []
+    if evaluation.course.evaluations.count() > 1:
+        evaluation.course.distribution = calculate_average_course_distribution(evaluation.course)
+        evaluation.course.avg_grade = distribution_to_grade(evaluation.course.distribution)
+        course_evaluations = [evaluation for evaluation in evaluation.course.evaluations.filter(state="published") if evaluation.can_user_see_evaluation(request.user)]
+        if request.user.is_reviewer:
+            course_evaluations += evaluation.course.evaluations.filter(state__in=['in_evaluation', 'evaluated', 'reviewed'])
+        course_weight_sum = sum(evaluation.weight for evaluation in course_evaluations)
+        for course_evaluation in course_evaluations:
+            course_evaluation.course_weight_sum = course_weight_sum
+            if course_evaluation.is_single_result:
+                course_evaluation.single_result_rating_result = get_single_result_rating_result(course_evaluation)
+            else:
+                course_evaluation.distribution = calculate_average_distribution(course_evaluation)
+                course_evaluation.avg_grade = distribution_to_grade(course_evaluation.distribution)
+            all_evaluations_of_course.append(course_evaluation)
```

Comment:
This is one more place where grade calculation is happening now, I feel like we want this in one central method somehow.

Reply:
true. similar to the visibility checks this should somehow be centralized. there could be a method filtering a list of evaluations to those a certain user can see and adding the appropriate grading infos to them or something like that.